src/mock_server/bare_server.rs

```diff
@@ -190,6 +210,16 @@ impl MockGuard {
         let (mounted_mock, _) = &state.mock_set[self.mock_id];
         mounted_mock.received_requests()
     }
+
+    pub async fn is_satisfied(&self) -> bool {
```

Comment:
We need docs here, ideally with an example (even in the same one you added in `MockGuard`'s main docs).